test/src/test/java/org/corfudb/integration/CorfuReplicationClusterConfigIT.java

```diff
@@ -0,0 +1,618 @@
+                standbyRuntime.getAddressSpaceView().getLogTail());
+
+        // Write 5 entries to active map
+        for (int i = firstBatch; i < secondBatch; i++) {
+            activeRuntime.getObjectsView().TXBegin();
+            mapActive.put(String.valueOf(i), i);
+            activeRuntime.getObjectsView().TXEnd();
+        }
+        assertThat(mapActive.size()).isEqualTo(secondBatch);
+
+        // Wait until data is fully replicated again
+        waitForReplication(size -> size == secondBatch, mapStandby, secondBatch);
+        log.info("After delta sync, both maps have size {}. Current active corfu[{}] log tail " +
+                        "is {}, standby corfu[{}] log tail is {}", secondBatch, activeClusterCorfuPort,
+                activeRuntime.getAddressSpaceView().getLogTail(), standbyClusterCorfuPort,
+                standbyRuntime.getAddressSpaceView().getLogTail());
+
+        // Verify data
+        for (int i = 0; i < secondBatch; i++) {
+            assertThat(mapStandby.containsKey(String.valueOf(i))).isTrue();
+        }
+        log.info("Log replication succeeds without config change!");
+
+        // Perform a role switch
+        corfuStore.tx(DefaultClusterManager.CONFIG_NAMESPACE)
+                .update(DefaultClusterManager.CONFIG_TABLE_NAME, DefaultClusterManager.OP_SWITCH,
+                        DefaultClusterManager.OP_SWITCH, DefaultClusterManager.OP_SWITCH)
+                .commit();
+        assertThat(configTable.count()).isOne();
+
+        // Write 5 more entries to mapStandby
+        for (int i = secondBatch; i < thirdBatch; i++) {
+            standbyRuntime.getObjectsView().TXBegin();
+            mapStandby.put(String.valueOf(i), i);
+            standbyRuntime.getObjectsView().TXEnd();
+        }
+        assertThat(mapStandby.size()).isEqualTo(thirdBatch);
+
+        // Wait until data is fully replicated again
+        waitForReplication(size -> size == thirdBatch, mapActive, thirdBatch);
+        log.info("Data is fully replicated again after role switch, both maps have size {}. " +
+                        "Current active corfu[{}] log tail is {}, standby corfu[{}] log tail is {}",
+                thirdBatch, activeClusterCorfuPort, activeRuntime.getAddressSpaceView().getLogTail(),
+                standbyClusterCorfuPort, standbyRuntime.getAddressSpaceView().getLogTail());
+
+        // Double check after 10 seconds
+        TimeUnit.SECONDS.sleep(mediumInterval);
+        assertThat(mapActive.size()).isEqualTo(thirdBatch);
+        assertThat(mapStandby.size()).isEqualTo(thirdBatch);
+    }
+
+    /**
+     * This test verifies config change with a role switch during a snapshot sync transfer phase.
+     * <p>
+     * 1. Init with corfu 9000 active and 9001 standby
+     * 2. Write 50 entries to active map
+     * 3. Start log replication: Node 9010 - active, Node 9020 - standby
+     * 4. Perform a role switch with corfu store
+     * 5. Standby will drop messages and keep size 0
```

Comment:
Actually, role is not expected to change when in transfer phase.  Only when we report transfer as 100%, the flip happens.  Will have to decide how we want to handle this case.

Reply:
Yeah, these two cases are not very clear right now. We did know much about workflow when designed these two cases.